### ISSUE TITLE
The mod in its default configuration prevents campaign dialog from triggering after completion of objectives

### ISSUE THREAD

**Issue by artyom-zuev** [m22spencer/BattletechPerformanceFix#37]
Two easiest to reproduce situations you can test this in are "Return to your employers' platform" objective in the very first story contract and "Secure the Argo" objective in the Argo story mission - moving all the mechs into the target zone fails to trigger the dialog continuing the mission if this mod is active.

**Comment by m22spencer** [comment]
Agreed, ContractLagFix is likely the culprit.  Will investigate further.

**Comment by nuvix** [comment]
Hi all, I had a problem with the mission never ending and staying stuck in combat mode. Not sure if it's related to this but it might be. I tried turning off ContractLagFix and ShopTabLagFix but the problem still occured. In the end I found it was a different mod called CBTMovement which causes you to enter combat mode directly and thus never trigger any cut scenes.

**Comment by valerian** [comment]
I can confirm that the issue is still present for me, and disabling ContractLagFix solves it.

If this can be of any use for someone else experiencing this issue as well: I noticed that a mission saved with ContractLagFix enabled will work properly when loaded again without ContractLagFix, so no need to restart it.

**Comment by xeryx01** [comment]
This issue still exists and very prominent, how this has not been fixed yet is beyond me.  It happens particularly on take base mission, when all objectives are complete, the missions will NOT complete the mission. I have turned everything off that I thought could cause this. PLEASE fix this issue. 
[BTPF 181 Settings and LOG.zip](https://github.com/m22spencer/BattletechPerformanceFix/files/4252406/BTPF.181.Settings.and.LOG.zip)
[output_log.txt](https://github.com/m22spencer/BattletechPerformanceFix/files/4252408/output_log.txt)
[ModTek_runtime_log.txt](https://github.com/m22spencer/BattletechPerformanceFix/files/4252409/ModTek_runtime_log.txt)
[harmony_summary.log](https://github.com/m22spencer/BattletechPerformanceFix/files/4252410/harmony_summary.log)
[ModTek.log](https://github.com/m22spencer/BattletechPerformanceFix/files/4252411/ModTek.log)

**Comment by m22spencer** [comment]
I've never been able to reproduce this, and even after having the verify fix on in the big modpacks for several weeks have not gotten a single report of it firing.

Thanks for the logs, but you have a bunch of other mods enabled. I'm not going to try to hunt this down based on that.

"It is most likely that lance-def stuff."
I'm not sure what you're referring to?

"It needs to be able to be turned off! OFF!"
Turn it off then. That's why there is a settings.json file.

**Comment by cepreu2github** [comment]
Happens to me too. But only if I had loaded save file made in tactical prior to completing objectives. If am playing mission from begining till end without any save&load everything works smoothly. But if I ever load mission save (not global map save) its objectives can't be completed at all. I can confirm that disabling ContractLagFix fixes problem for me as well.

**Comment by cepreu2github** [comment]
And obviously problem is not in saving, but in loading. I had a save made in pure vanilla without any mods installed. After putting my mods folder back without ContractLagFix overriden to false, I couldn't able to complete the mission. Simplest way to test: just start a fresh campaign, make a save at coronation mission as soon as button become available. Load it and try to complete objective. It's not working properly anymore.

**Comment by m22spencer** [comment]
Ah, that makes sense.  RogueTech disables combat saves, so that's why no reports.

Value is cached in the constructor, which means the load deserializer must not be calling it.

For now, I would just disable the option in the config.

**Comment by Hobbes74** [comment]
I've just been pointed to this thread. I've had several occurances of this issue while using this mod with the Hyades Rim in the past two years while using BPF.

**Comment by thieman** [comment]
This really should be pinned or something, I just started the game and installed this on recommendations from the community. Made the first two campaign missions randomly become unfinishable, terrible experience.
